templates/template.yaml

```diff
@@ -136,15 +136,38 @@ Parameters:
     Description: Comma-delimited list of subnets to deploy Fargate tasks in. This parameter must be set if the DeployVpc parameter is "false", otherwise it is ignored.
     Type: CommaDelimitedList
     Default: ""
+  DeployWebUI:
+    Description: Specify whether you would like a Web User Interface deployed. (AuthMethod must be Cognito if WebUI is deployed).
+    Type: String
+    Default: "true"
+    AllowedValues:
+      - "true"
+      - "false"
+  AuthMethod:
+    Description: Authentication method to use for the API (and WebUI if deployed).
+    Type: String
+    Default: "Cognito"
+    AllowedValues:
+      - "Cognito"
+      - "IAM"
+
+Rules:
+  ValidateAuth:
+    RuleCondition: !Equals [!Ref DeployWebUI, "true"]
+    Assertions:
+      - AssertDescription: IAM Auth cannot be chosen when deploying the WebUI.
+        Assert: !Not [!Equals [!Ref AuthMethod, "IAM"]]
```

Comment:
in the interest of future proofing for other auth methods, should this check be checking that AuthMethod === Cognito?

Reply:
Good point, however one of the next changes we plan is to add Cognito with mandatory MFA as an option. So at that point we would have more than one Cognito option so i choose this != way of checking.

Reply:
the implementation of that might not necessarily involve a new AuthMethod. for example it could be a separate param. even so you could use the !Or operator in that scenario

Reply:
Fair point, i had not considered doing it with !Or, thanks.

Reply:
Logic updated to match your suggestion